### PR TITLE
cleanup gateway SDS documentation

### PR DESCRIPTION
**Pull request by rshriram**
cc @frankbu

### CODE REVIEW

File: networking/v1beta1/gateway.proto [hunk 1]
```diff
@@ -625,22 +625,13 @@ message Server {
...
-    // credentialName-cacert. The semantics of the name are platform
-    // dependent.  In Kubernetes, the default Istio supplied credential
-    // server expects the credentialName to match the name of the
-    // Kubernetes secret that holds the server certificate, the private
-    // key, and the CA certificate (if using mutual TLS). Set the
-    // `ISTIO_META_USER_SDS` metadata variable in the proxy to
-    // enable the dynamic credential fetching feature.
+    // For gateways running on Kubernetes, the name of the secret that
+    // holds the TLS certs including the CA certificates. Applicable
+    // only on Kubernetes. The secret (of type `generic`) should
```
Comment (frankbu) [inline networking/v1beta1/gateway.proto]:
```suggestion
    // only on Kubernetes and only if the dynamic credential fetching
    // feature is enabled by setting the `ISTIO_META_USER_SDS`
    // metadata variable in the proxy. The secret (of type `generic`) should
```

File: networking/v1beta1/gateway.proto [hunk 2]
```diff
@@ -625,22 +625,13 @@ message Server {
...
+    // contain the following keys and values: `key: <privateKey>`,
+    // `cert: <serverCert>`, `cacert: <CACertificate>`. Set the
```
Comment (frankbu) [inline networking/v1beta1/gateway.proto]:
```suggestion
    // `cert: <serverCert>`, `cacert: <CACertificate>`.
```

File: networking/v1beta1/gateway.proto [hunk 3]
```diff
@@ -625,22 +625,13 @@ message Server {
...
+    // `ISTIO_META_USER_SDS` metadata variable in the proxy to enable
+    // the dynamic credential fetching feature.
```
Comment (frankbu) [inline networking/v1beta1/gateway.proto]: delete

**Comment by rshriram** [comment]
@frankbu done.

**Comment by frankbu** [comment]
@rshriram LGTM, but it seems that now istio/api needs 2 code owner approvals, and I'm not even one any more. Maybe I can get my code owner privilege back?

**Comment by rshriram** [comment]
> LGTM, but it seems that now istio/api needs 2 code owner approvals, and I'm not even one any more. Maybe I can get my code owner privilege back?
I think what has happened is that someone has reset all the code owners to just the TOC for everything :(. Do you know what the status was before?

**Comment by frankbu** [comment]
Well I used to be code owner for istio/istio.io and istio/api (user facing things), but now just istio.io. Seems I still should also be for istio/api, no?

**Pull request merged**